<launch>

  <!-- <arg name="interface" default="can0" /> --> <!-- HH_230308 -->
  <arg name="receiver_interval_sec" default="0.01" />
  <arg name="sender_timeout_sec" default="0.01" />
  <arg name="enable_can_fd" default="false" />

  <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_receiver.launch.py">
    <arg name="interface1" value="can0" />
    <arg name="interface2" value="can1" /> 
    <arg name="interval_sec" value="$(var receiver_interval_sec)" />
    <arg name="enable_can_fd" value="$(var enable_can_fd)" />
  </include>

  <!-- <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_receiver.launch.py">
    <arg name="interface" value="can1" />
    <arg name="interval_sec" value="$(var receiver_interval_sec)" />
    <arg name="enable_can_fd" value="$(var enable_can_fd)" />
  </include> -->

  <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_sender.launch.py">
    <arg name="interface" value="can0" />
    <arg name="timeout_sec" value="$(var sender_timeout_sec)" />
    <arg name="enable_can_fd" value="$(var enable_can_fd)" />
  </include>

  <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_sender.launch.py">
    <arg name="interface" value="can1" />
    <arg name="timeout_sec" value="$(var sender_timeout_sec)" />
    <arg name="enable_can_fd" value="$(var enable_can_fd)" />
  </include>

</launch>
